docs/api/microcosm.md

```diff
@@ -5,13 +5,58 @@
 
 ## Overview
 
-Microcosm is [Flux](https://facebook.github.io/flux/) with first-class
-actions and state sandboxing.
+The Microcosm class provides a centralized place to store application
+state, dispatch actions, and track changes.
 
-The source of truth in Microcosm is a historical record of actions. As
-they move through a set lifecycle, Microcosm reconciles actions in the
-order they were created. This makes optimistic updates, cancellation,
-and loading states much simpler. They self clean.
+### Creating a Microcosm
+
+All Microcosm apps start by instantiating a Microcosm class. We call
+this instance of Microcosm a "repo":
+
+```javascript
+let repo = new Microcosm()
+```
+
+### Options
+
+The first argument of the Microcosm constructor is an object of
+options:
+
+```javascript
+let repo = new Microcosm({ maxHistory: 10 })
+```
+
+Microcosm supports the following options:
+
+1. `maxHistory:number`: In Microcosm, data is changed by responding
+   to [actions](./actions.md). This builds up a history that can be
+   useful for debugging and undo/redo behavior. By default, Microcosm
+   gets rid of any old actions to reduce memory usage. By setting
+   `maxHistory`, you can tell Microcosm to hold on to those actions.
+2. `updater:function`: When specified, this allows you to configure
+   how Microcosm releases change events. This is useful for batching
+   together frequent state changes. See
+   the [Batch Updates](../recipes/batch-updates.md) recipe for more
+   information.
+
+Feel free to add additional options to fit your use case. Any options
+you provide to Microcosm are passed into the `setup` lifecycle method:
+
+```javascript
+import Autosave from './effects/autosave'
```

Comment:
can this example code demonstrate the concept without introducing effects? to keep the section as simple as possible